docker/Dockerfile

```diff
@@ -0,0 +1,15 @@
+FROM node:22.4.0-slim
+
+# Set a non-root user to run the container
+RUN useradd --create-home --shell /bin/bash mesh
+USER mesh
+
+WORKDIR /app
+
+COPY package.json ./
+
+RUN npm i --only=production
+
+EXPOSE 4000
+
+ENTRYPOINT ["./node_modules/.bin/mesh-serve"]
```

Comment:
Should we wrap this with `node` and use `CMD`? what's the reason for picking `ENTRYPOINT`?

Reply:
oh, this way you can pass more args directly from Docker?

Reply:
the usage of `ENTRYPOINT` is to allow appending arguments into the `mesh-serve` command like `--supergraph`